### ISSUE TITLE
Remarques sur le catalogue d'objets vprojet_2023-11_ChampApplication / ConditionUnitaireSpécifique

### ISSUE THREAD

**Issue by ynakhimo** [cnigfr/structuration-reglement-urbanisme#17]
## Classes ChampApplication et ConditionUnitaireSpécifique
- Il semble que ChampApplication permet de définir une sorte de BandeConstructible à partir de la limite de référence, sur une _profondeur_ donnée (par ailleurs ladite _profondeur_ devrait pouvoir être mesurée à partir de la limite de référence plutôt qu’uniquement des bordures donnant sur la voirie), tandis que ConditionUnitaireSpécifique gère les périmètres issus des Prescriptions Graphiques (PG). Cependant parmi les références de ChampApplication on retrouve des intersections avec PG - ce n’est pas clair comment faire le lien avec les PG dans ce cas. Ou bien avec d’autres classes, telles que RouteBordante, qui permettraient d’identifier les limites de référence qui génèrent un ChampApplication de la règle. De manière générale, on pourrait peut-être simplifier les choses en ayant une seule classe à la place des deux, qui peut être liée à elle-même (e.g. pour les intersections) ou implémentée par d'autres classes telle que BandeConstructible ?
- Il est écrit que ChampApplication est associé aux conditions, cela veut-il dire que l’on ne peut pas ajouter un ChampApplication sans ajouter une ConditionUnitaire ? Par exemple, si la règle limite la hauteur au-delà d’une BandeConstructible de 20m, le fait de restreindre cette règle au périmètre de la BC implique déjà une condition sur l’existence de cette dernière. Est-il vraiment pertinent de la rendre explicite ? De plus, en théorie pour la même ConditionUnitaire, il peut y avoir plusieurs ChampApplication et ContrainteUnitaire, ce qui serait un autre argument pour les dissocier.
- la classe est parfois mentionné en tant que ContrainteUnitaireSpécifique

**Comment by GT-CNIG-DDU** [comment]
> ladite _profondeur_ devrait pouvoir être mesurée à partir de la limite de référence plutôt qu’uniquement des bordures donnant sur la voirie

Effectivement, dans le projet de standard, remplacer "Détermine la profondeur par rapport à la référence des bordures donnant sur la voirie." par  : "Détermine la profondeur par rapport à la référence."

> parmi les références de ChampApplication on retrouve des intersections avec PG - ce n’est pas clair comment faire le lien avec les PG dans ce cas.

Effectivement, enlever "PG" des valeurs de l’énumération `TypeReference` puisque la définition de la classe indique bien : _"S’il s’agit de prescription graphique, le lien entre la prescription d’urbanisme sectorisée et la ConditionUnitaire ne se fait pas avec le champ d’application mais à travers la classe contrainteUnitaireSpécifique faisant directement le lien avec la classe Prescription du standard CNIG PLU."_

> la classe est parfois mentionné en tant que ContrainteUnitaireSpécifique

Préciser où, pour cibler les corrections.

**Le nom de la classe est à corriger dans le [modèle de données graphique](https://github.dev/cnigfr/structuration-reglement-urbanisme/tree/master/standard_niveau_2)** : remplacer `perimetreUnitaire` par `champApplication`.

![image](https://github.com/cnigfr/structuration-reglement-urbanisme/assets/79853274/e638c179-74e1-439a-9e47-2144f5294e40)

**Comment by alisonlenain** [comment]
Revoir le lien avec entre la classe ChampApplication et PG avec Buildzr. 
Proposition de supprimer l'énumérant "intersectionTerrainPG" . 
Revoir l'attribut profondeur ?

**Comment by mbrasebin** [comment]
Dans le fond, c'est exactement le même sujet que pour les retraits/alignements. 

Est-ce que pour l'usager il est plus simple d'avoir 2 classes avec des intitulés explicites ou une seule avec beaucoup d'attributs qui ne vont sans doute pas tous s'activer ? Dans l'idée, si le but set de faire le lien avec une PG, on ajoutera sans doute pas en plus une bande de constructibilité (et même pour se cas très rare, le chainage de conditions autorisera de modéliser cela).

Pour moi c'est vraiment cette question de fond de choix de modélisation qui est importante et doit être cohérente sur l'ensemble du standard.  Moi je millite au maximum pour la sépration en différentes classes, lorsque l'activation de tous les attributs de la classe mène à une situation qui n'est modélise une idée qui n'est pas unitaire (comme le nom de la classe mère l'indique) ou incohérente.

**Comment by GT-CNIG-DDU** [comment]
Je partage l'avis de @mbrasebin. Le risque (à éviter également!) en multipliant les classes d'objets est bien sûr d'élaborer un modèle très complexe qui ressemblera à une usine à gaz. Mais au moins : développer le modèle lui permet d'être évolutif et plus lisible qu'un modèle avec des classes "factorisantes" qui s'avèrent finalement être des classes abstraites.
Nous parlons là pour l'usager concepteur et/ou développeur d'applications basées sur le modèle, car pour l'utilisateur final nous espérons bien que l'interface homme machine permettra de rendre tout cela "transparent".

@ynakhimo : a-t-on traité ton deuxième point ? Je n'en ai pas l'impression... Dans ce cas, ne vaut-il pas mieux créer une issue dédiée ?
Il me semble que @alisonlenain a traité le troisième point dans la rédaction du projet de standard.

**Comment by alisonlenain** [comment]
Modification de la définition de l’attribut profondeur : "Détermine la profondeur par rapport à la référence."

Suppression de "PG" des valeurs de l’énumération TypeReference.

Voir le lien entre les classes ChampApplication avec BC et VoirieBordante ==> lien grace à une clé secondaire ? (Ajouter dans la classe ChampApplication : l'identifiant de la BC ou de la PG ( clé secondaire ) : nom de cet attribut à trouver et faire valider le SG6 pour la prochaine réunion.)

Traiter également : "il est écrit que ChampApplication est associé aux conditions, cela veut-il dire que l’on ne peut pas ajouter un ChampApplication sans ajouter une ConditionUnitaire ? Par exemple, si la règle limite la hauteur au-delà d’une BandeConstructible de 20m, le fait de restreindre cette règle au périmètre de la BC implique déjà une condition sur l’existence de cette dernière. Est-il vraiment pertinent de la rendre explicite ? De plus, en théorie pour la même ConditionUnitaire, il peut y avoir plusieurs ChampApplication et ContrainteUnitaire, ce qui serait un autre argument pour les dissocier".

**Comment by alisonlenain** [comment]
Lien entre les classes ChampApplication avec BC et VoirieBordante ==>  grace à une clé secondaire. Ajouter dans la classe ChampApplication : l'identifiant de la BC ou de la PG ( clé secondaire ) : nom de cet attribut à trouver et faire valider le SG6 pour la prochaine réunion.

**Comment by alisonlenain** [comment]
Suppression de l'attribut profondeur dans la classe ChampApplication (cf issue #19)

**Comment by GT-CNIG-DDU** [comment]
Dans le CR SG6 du 18/04/2024 l'issue est indiquée comme finalisée mais qu'en est-il de ce ce lien prévu entre classes ?  : 

> Lien entre les classes ChampApplication avec BC et VoirieBordante ==> grace à une clé secondaire. Ajouter dans la classe ChampApplication : l'identifiant de la BC ou de la PG ( clé secondaire ) : nom de cet attribut à trouver et faire valider le SG6 pour la prochaine réunion.
